docs/eventing/sugar/README.md

```diff
@@ -0,0 +1,357 @@
+use-cases:
+
+- When a Namespace is labeled with `eventing.knative.dev/injection=enabled`, the
+  controller will create a default broker in that namespace.
+- When a Trigger is annotated with `eventing.knative.dev/injection=enabled`, the
+  controller will create a Broker named by that Trigger in the Trigger's
+  Namespace.
+
+When a Broker is deleted and the above labels or annotations are in-use, the
+Sugar Controller will automatically recreate a default Broker.
+
+### Namespace Examples
+
+Creating a "default" Broker when creating a Namespace:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: example
+  labels:
+    eventing.knative.dev/injection: enabled
+EOF
+```
+
+To automatically create a Broker after a namespace exists, label the Namespace:
+
+```shell
+kubectl label namespace default eventing.knative.dev/injection=enabled
+```
+
+If the Broker named "default" already exists in the Namespace, the Sugar
+Controller will do nothing.
+
+### Trigger Examples
+
+Create a Broker named by a Trigger (`spec.broker`) in the Trigger's Namespace:
+
+```shell
+kubectl apply -f - << EOF
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: hello-sugar
+  namespace: candyland
+  annotations:
+    eventing.knative.dev/injection: enabled
+spec:
+  broker: gumdrops
+  subscriber:
+    ref:
+      apiVersion: v1
+      kind: Service
+      name: event-display
+EOF
+```
+
+> _Note_: If the named Broker already exists, the Sugar controller will do
+> nothing, and the Trigger will not own the existing Broker.
+
+This will make a Broker called "gumdrops" in the Namespace "candyland", and
+attempt to send events to the "event-display" service.
+
+If the Broker of the given name already exists in the Namespace, the Sugar
+Controller will do nothing.
+
+## Automatic Trigger Creation
+
+The Sugar Controller can also watch for any resource that conforms to the duck
+type "Addressable", see [the defintion](TODO: add a link for addressable
+definition) or the [Discovery](TODO: link to the discovery documentation)
+ClusterDuckType
+[addressables.duck.knative.dev](https://github.com/knative-sandbox/discovery/blob/master/config/knative/addressables.duck.knative.dev.yaml).
+
+For any addressable that has been labeled with
+`eventing.knative.dev/autotrigger=enabled`, the Sugar Controller will manage the
+Trigger(s) for it.
+
+To manage one or more triggers automatically, provide the annotation:
+
+```
+annotations:
+  autotrigger.eventing.knative.dev/filter.attributes: |
+    [{"type":"cloudevents.event.type", "custom-attribute":"match-value"}]
+```
+
+`autotrigger.eventing.knative.dev/filter.attributes` is a JSON array of any map
+of string to string. Any valid CloudEvents attribute name and value are allowed.
+For each map in the array, a new Trigger will be created. If this property is
+omitted, a Trigger with no filters will be created for the Addressable.
```

Comment:
```suggestion
omitted, a Trigger with no filters will be created for the Addressable resource.
```
I don't know enough to know if Addressable is a resource type, or if it's supposed to still have "resource" after it. Added the suggestion to match "Addressable resource" as it shows up below in the same way.